src/control/cmd/daos/pool.go

```diff
@@ -276,6 +278,100 @@ func (cmd *poolQueryCmd) Execute(_ []string) error {
 	return nil
 }
 
+type poolQueryTargetsCmd struct {
+	poolBaseCmd
+
+	Rank      uint32 `long:"rank" required:"1" description:"Engine rank of the targets to be queried"`
+	Targetidx string `long:"target-idx" description:"Comma-separated list of target idx(s) to be queried"`
+}
+
+func convertDaosSpaceInfo(in *C.struct_daos_space, mt C.uint) *mgmtpb.StorageTargetUsage {
+	if in == nil {
+		return nil
+	}
+
+	return &mgmtpb.StorageTargetUsage{
+		Total: uint64(in.s_total[mt]),
+		Free:  uint64(in.s_free[mt]),
+	}
+}
+
+// For using the pretty printer that dmg uses for this target info.
+//
+// Does the same thing as ds_mgmt_drpc_pool_query_target() will do, to stuff the target info
+// into a protobuf message, then use the automatic conversion from proto to control.
+
+func convertPoolTargetInfo(ptinfo *C.daos_target_info_t) (*control.PoolQueryTargetInfo, error) {
```

Comment:
Hmm. For the record, this is kind of a hack. I know that I set the precedent with the pool info stuff, but it was intended as a shortcut to finish the porting of the daos tool frontend to Go rather than trying to establish best practices. The main problem with using the protobuf stuff and pretty-printers is that it pulls in dependencies that cause the compiled `daos` binary to be much larger than it really needs to be otherwise.

That having been said, I haven't had the cycles to go back and fix the existing tech debt yet so I won't stand in the way of this landing as-is for now. I just wanted to point this out as something that will be fixed later in a way that preserves existing functionality while removing those extra dependencies.